src/apiClient/asset.ts

```diff
@@ -25,8 +25,8 @@ export abstract class UploadRequest<MT = {}> {
    * @param file File object to upload.
    * @param model Optional instance of related model that this asset is associated with.
    */
-  protected abstract async prepareUploadRequest(file: File, model?: MT): Promise<PrepareUploadResponse>
-  protected abstract async prepareUploadRequest(file: File): Promise<PrepareUploadResponse>
```

Comment:
Why remove `async`?

Reply:
Because Typescript, from version 4.1, started to disallow the abstract class to don't contain modifiers as `async` , and delegate this to the children who extend it.
* [Here the official docs](https://stackoverflow.com/questions/65355164/error-ts1243-async-modifier-cannot-be-used-with-abstract-modifier)
* [Here more details from SO](https://stackoverflow.com/questions/65355164/error-ts1243-async-modifier-cannot-be-used-with-abstract-modifier)